Append unified to cgroup filepath in v2 case only if exists

markrogers95 opened the pull request:
I think that in the case that cgroupsv2 is not in use, the `/sys/fs/cgroup/unified` file isn't expected to exist meaning we get a file stat error when trying to complete the operation. 

I'll caveat with I'm not _too_ au fait with how this logic is used but I can see in testing that we fail to attach the tracer at this point based on the stat error.

markrogers95 commented:
seems to have systematically broken the tests here, must run deeper than this initial line change - will have a look

markrogers95 commented:
Hey @rafaelroquetto, thanks for letting me know! Please do give me a shout when all fixed :+1:

rafaelroquetto commented:
Hey @markrogers95 the tests are back in action - I tried to give this PR a few runs but they are still failing - I believe your changing is having unintended side-effects.

markrogers95 commented:
Hey @rafaelroquetto, I've updated this a little now - it still _feels_ wrong to have the `enabled` condition this way around, but it's clear that it was affecting the tests in some way I agree. I've changed approach so as to keep the logic as it is, but before altering the cgroup path for `sock ops` I'll stat the file and only update if it exists. This seems to keep the tests happy and get around the issue with my limited knowledge of the code!

grcevski reviewed (approved):
Nice catch! LGTM!

grcevski commented on pkg/internal/ebpf/instrumenter.go:
```diff
@@ -431,6 +431,10 @@ func getCgroupPath() (string, error) {
 
 	enabled, err := v2.Enabled()
 	if !enabled {
+		if _, pathErr := os.Stat(filepath.Join(cgroupPath, "unified")); pathErr != nil && os.IsNotExist(pathErr) {
```
Actually, I am wondering if we should return in case of any error?

grcevski replied:
What I mean, does it make sense to check if it's not found error? If we couldn't stat the file it will likely mean some sort of permission issue, in which case it might make sense to bail?

grcevski commented:
> @grcevski thanks! if/when this goes in, it would be great to get a release containing this and the bpf_loop override so we can test in a real environment too - is there a process around this please?

Yeah sounds good, we'll release 2.1.0 with these fixes soon.

grcevski commented:
Thanks again for your work on this here @markrogers95 !

markrogers95 commented:
amazing, that sounds great - thanks very much again!
